### DISCUSSION TITLE
Typescript issue

### DISCUSSION THREAD

**Discussion by ergi98** [trpc/trpc#2040] [Q&A] [answered]
Hi everyone!
I recently started building a project using tRPC and Express for my backend and React for the frontend.

I am having trouble with using trpc in my client. The issue is that for some reason the types are not inferred and all the data being returned is typed as "any".

I need to point out that the project is not a monorepo but I have installed the backend code as a dependency to the client project.

Thank you in advance!

Usage in client:

<img width="835" alt="Screenshot 2022-06-24 at 23 24 48" src="https://user-images.githubusercontent.com/70513280/175692666-6ffd3899-eea2-40d4-9c83-70d021b81597.png">

index.ts of server:

<img width="835" alt="Screenshot 2022-06-24 at 23 32 03" src="https://user-images.githubusercontent.com/70513280/175695911-d6022c0b-fd88-4fc9-9383-30b7314577be.png">

context of trpc:

<img width="880" alt="Screenshot 2022-06-24 at 23 32 56" src="https://user-images.githubusercontent.com/70513280/175696004-8b6307d3-923b-4c10-8405-bab59de4f804.png">

the route i am calling:

<img width="880" alt="Screenshot 2022-06-24 at 23 33 30" src="https://user-images.githubusercontent.com/70513280/175696021-8cb4e124-16ef-4fe6-8fa1-500a2f51df7f.png">

the model of the data:

<img width="786" alt="Screenshot 2022-06-24 at 23 33 51" src="https://user-images.githubusercontent.com/70513280/175696050-da8ccc75-aa40-4b23-a8ec-a98f44ec00e4.png">

**Comment by ludwigbacklund** [comment, marked as answer]
What does your `@utils/trpc` look like?
Are you creating your react hooks by passing the type of your appRouter like this? 👉 https://github.com/trpc/examples-next-prisma-starter/blob/main/src/utils/trpc.ts#L27

**Reply by ergi98** [reply]
I gave up all together and joined both of my projects in a monorepo. Works like a charm!

**Reply by cl0ckwork** [reply]
love the idea of tRPC, but im struggling to get the types to play nicely in the client.

is this the only solution to symptoms like like those described above?  essentially tRPC must be used within a mono repo? must the tsconfigs and ts versions match exactly as well?

ive noticed the types don't work using the following 

```ts
// client/trpc.ts
import type { AppRouter } from '../../some/backend/monorepo/src/trpc/appRouter.ts'
export const trpc = createTRPCReact<AppRouter>();
```

not only does typescript on the client yell about all sorts of backend type errors and bad paths, but the types yielded by `something.path.userQuery()` are all `any`. 🤔 

```ts
// server/appRouter.ts
export type AppRouter = typeof appRouter;
```

**Reply by cl0ckwork** [reply]
for those looking, i found this [thread](https://github.com/trpc/trpc/discussions/1860#discussioncomment-4406931)
